applications/GeoMechanicsApplication/custom_elements/transient_thermal_element.h

```diff
@@ -192,6 +193,16 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) TransientThermalElement : public Ele
                                                           << Id() << std::endl;
     }
 
+    void CheckSaturatedLaw() const
+    {
+        const auto& r_variable = RETENTION_LAW;
+        KRATOS_ERROR_IF_NOT(GetProperties().Has(r_variable)) << r_variable.Name()
+                                                            << " does not exist in the thermal element's properties" << std::endl;
+        KRATOS_ERROR_IF_NOT(GetProperties()[r_variable] == "SaturatedLaw") << r_variable.Name() << " has an invalid law ("
+                                                           << GetProperties()[r_variable] << ") at element "
+                                                           << Id() << std::endl;
+    }
+
```

Comment:
If you accept using         CheckProperty(RETENTION_LAW, "SaturatedLaw" );
then
void CheckProperty(const Kratos::Variable<std::string>& rVariable, const std::string & rName ) const
    {
        KRATOS_ERROR_IF_NOT(GetProperties().Has(rVariable)) << rVariable.Name()
                                                            << " does not exist in the thermal element's properties" << std::endl;
        KRATOS_ERROR_IF_NOT(GetProperties()[rVariable] == rName) << rVariable.Name() << " has a value of (" << GetProperties()[rVariable]
                                                                 << ") instead of (" << rName << ") at element "
                                                                 << Id() << std::endl;
    }

Reply:
I like this idea, copied your code adding <> around std::string.